Cypress setup so the frontend can be tested

martinj2001 opened the pull request:
**Description**

Couple of js config files as well as a folder for our Cypress tests. 

Fixes #3302

Felienne commented:
Thanks @martinj2001!

We ask contributors to include a **How to test** section in each PR, can you add one? I think the PR template should have created one for you (let me know if it did not)

If nothing should be changed, that too can be added as a How to test! But in this form, it is not clear to me what I should to verify that this PR is ok. Should I expect a change? If so, what change?

Felienne commented:
Based on the contents of the PR, I suspect I should be able to run this initial test? If so, maybe we want to: 

1. Add this to the docs so people know how to run the tests (and what to install?)
2. Add this to the build script so it runs automatically on each PR

martinj2001 commented:
@Felienne I updated Contributing.md in the following commit: 7316038f40a2d8bbbed98c7b057a470f4c9becdf.
I hope this makes clear how to run the first front-end test. We are still investigating how to execute all tests on the command line, so we will add that in the future!

TiBiBa commented on CONTRIBUTING.md:
````diff
@@ -58,6 +64,16 @@ To run the unit tests:
 
+To run the front-end tests:
+
+```bash
+$ cd /your/project/path
+$ npx cypress open
````
We can safely assume that contributing use a virtual environment when working with Hedy. This can be replaced by simply stating `npx cypress open`. Did you add `cypress` to the requirements.txt as well?

martinj2001 replied:
Yes, you're right. I changed that!

TiBiBa reviewed (changes requested):
Looks almost fine to me! However, I would suggest to move the `/cypress` folder inside the `/tests` folder we already have. This makes the project structure a but more clean.

martinj2001 commented:
@Felienne @TiBiBa I fixed all the problems, moved cypress to the /tests folder and edited contributing.md so that it matches the new structure. We will look into automatic front-end testing, for now this is basic file structure for our group to use.

martinj2001 commented:
I also fixed one error in the app.ts file. It came up during the unit testing

Felienne commented on CONTRIBUTING.md:
````diff
@@ -58,6 +64,16 @@ To run the unit tests:
+$ npx cypress open
+```
+
+You will see the Cypress Launchpad in which you can choose to open the End2End testing panel. Afterwards you are able to run all the tests configured in the test suite, as well as adding your own according to [the documentation of Cypress](https://docs.cypress.io/guides/end-to-end-testing/writing-your-first-end-to-end-test).
````
```suggestion
You will see the Cypress Launchpad in which you should choose to open the End2End testing panel. Afterwards you are able to run all the tests configured in the test suite, as well as adding your own according to [the documentation of Cypress](https://docs.cypress.io/guides/end-to-end-testing/writing-your-first-end-to-end-test).
```

Felienne reviewed (approved):
This seems to work team!!

<img width="512" alt="image" src="https://user-images.githubusercontent.com/1003685/191234811-3313c181-4cc1-4e60-94a6-400fe96b2cdf.png">

Well done and congrats on the first PR merged!
